packages/flutter/lib/src/widgets/async.dart

```diff
@@ -471,6 +471,9 @@ class StreamBuilder<T> extends StreamBuilderBase<T, AsyncSnapshot<T>> {
        super(key: key, stream: stream);
 
   /// The build strategy currently used by this builder.
+  ///
+  /// This builder can get called arbitrarily and should not do anything beyond
+  /// building a widget.
```

Comment:
Should we just say:

This builder must only return a widget and should not have any side effects as it may be called multiple times.